### DISCUSSION TITLE
Photoprism keeps resetting

### DISCUSSION THREAD

**Discussion by PycraftDeveloper** [photoprism/photoprism#3715] [Q&A] [answered]
I recently setup photoprism on my TrueNAS Scale server using docker. I uploaded all my photos/videos to the server. I set-up custom logins for different people. I restarted my server and everything was fine.

Two weeks later I come back to access my photos, but all my photos (18,000+, ~26 GB of data) had gone, as had all the custom logins I had setup. I had other backups of the photos but photoprism was by far the easiest to access and use. Is there a way to recover the data from the server and prevent this from happening again?

(it has reset everything to factory settings by the looks of things)

![image](https://github.com/photoprism/photoprism/assets/81379254/6a727f90-1a34-4da9-99de-0ed4a4eaf818)
![image](https://github.com/photoprism/photoprism/assets/81379254/9b586df0-794a-42db-a9ef-c3001f0e6517)
![image](https://github.com/photoprism/photoprism/assets/81379254/1113876a-8ff1-425f-a28b-42efa4ec1e14)

**Comment by andrewdolphin** [comment]
How did you set it up with docker? It sounds like you may not have mounted the appropriate database and config files outside of the container so the data would be lost if the container restarted

**Reply by PycraftDeveloper** [reply]
Thank you for the response. I set it up through TrueNAS Scale's Apps section, and thought I had set-up appropriate config files outside of the container, as the app data is (or was) stored in a separate pool in a network accessible folder (through SMB). I am working on reinstalling photoprism and using a different tutorial to see if the problem goes away (as it was likely to be human error), I'll update this thread in a few days when I have done that!

**Reply by PycraftDeveloper** [reply, marked as answer]
Following on from this, I decided to reinstall Photoprism and removed the folders I had set up originally. Then I followed the exact same steps as before and its been a few weeks since the initial incident and everything has been working fine. I think it might have been a drive permission error (so photoprism wouldn't write to the disk) however the behaviour I observed with the initial issue doesn't fully back up that conclusion. TLDR everything seems to be working now!
